src/properties/mod.rs

```diff
@@ -62,6 +62,87 @@ pub mod text_property;
 /// Module for `UnknownProperty`
 pub mod unknown_property;
 
+/// The default read() implementation
+#[macro_export]
+macro_rules! read {
```

Comment:
this and the read_header macro should probably have an `impl_` prefix for them, as that makes it more clear what they do

Reply:
That's totally fair. I will make this change.